x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.ts

```diff
@@ -225,3 +230,70 @@ export const createIndexPipelineDefinitions = (
   });
   return { created: [indexName, `${indexName}@custom`, `${indexName}@ml-inference`] };
 };
+
+/**
+ * Format the body of an ML inference pipeline for a specified model.
+ * Does not create the pipeline, only returns JSON for the user to preview.
+ * @param modelId modelId selected by user.
+ * @param sourceField The document field that model will read.
+ * @param destinationField The document field that the model will write to.
+ * @param esClient the Elasticsearch Client to use when retrieving model details.
+ */
+export const formatMlPipelineBody = async (
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<ESPipeline> => {
+  const models = await esClient.ml.getTrainedModels({ model_id: modelId });
+  // if we didn't find this model, we can't return anything useful
+  if (models.trained_model_configs === undefined || models.trained_model_configs.length === 0) {
+    throw new Error(`Couldn't find any trained models with id [${modelId}]`);
+  }
+  const model = models.trained_model_configs[0];
+  // if model returned no input field, insert a placeholder
+  let modelInputField = 'MODEL_INPUT_FIELD';
+  if (
+    model.input !== undefined &&
+    model.input.field_names !== undefined &&
+    model.input.field_names.length > 0
+  ) {
+    modelInputField = model.input.field_names[0];
+  }
```

Comment:
I think you can make this conditional a little easier with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):
```node
> let model = {};
undefined

> model.input?.field_names?.length > 0
false

> model = { input: { field_names: [] } };
{ input: { field_names: [] } }

> model.input?.field_names?.length > 0
false

> model = { input: { field_names: [1] } };
{ input: { field_names: [ 1 ] } }

> model.input?.field_names?.length > 0
true
```

```suggestion
  const modelInputField = model.input?.field_names?.length > 0
    ? model.input.field_names[0]
    : 'MODEL_INPUT_FIELD';
```